designer/client/src/components/modals/AdvancedParametersSection.tsx

```diff
@@ -7,11 +7,20 @@ import AccordionDetails from "@mui/material/AccordionDetails";
 
 interface Props {
     componentId: string;
+    onChange?: (isExpanded: boolean) => void;
+    expanded?: boolean;
 }
 
-export function AdvancedParametersSection({ children, componentId }: PropsWithChildren<Props>): JSX.Element {
+export function AdvancedParametersSection({ children, componentId, onChange, expanded }: PropsWithChildren<Props>): JSX.Element {
     return (
-        <Accordion disableGutters elevation={0} sx={{ border: 0, "&::before": { display: "none" } }}>
+        <Accordion
+            disableGutters
+            elevation={0}
+            sx={{ border: 0, "&::before": { display: "none" } }}
+            onChange={onChange != null ? (_, isExpanded) => onChange(isExpanded) : null}
```

Comment:
Why is onChange optional? I can see that this component is called in one place, with the onChange property.

BTW:

this form

```
onChange={onChange && ((_, isExpanded) => onChange(isExpanded))}
```
or

```
onChange={((_, isExpanded) => onChange?.(isExpanded))}
```

is more readable

the `expanded` property  also could be required in this case

Reply:
Done